src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_cpu_blocking.cpp

```diff
@@ -83,11 +83,17 @@ bool BrgemmCPUBlocking::mark_blocking_loops(LinearIR& linear_ir,
     if (stand_alone(type))
         return res;
 
-    const auto copy_b_expr = linear_ir.get_expr_by_node(brgemm->get_brgemm_copy());
-    const ov::snippets::VectorDims full_subtensor(2, get_full_dim_value());
-    copy_b_expr->get_input_port_descriptor(0)->set_subtensor(full_subtensor);
-    copy_b_expr->get_output_port_descriptor(0)->set_subtensor(full_subtensor);
-
+    const auto copy_b_expr = repacking::get_copy_b_expr(brgemm_expr);
+    if (copy_b_expr) {
```

Comment:
Ok, I see. That's exactly why we need a descriptor that tells us that there must be a repacking op, but it was deleted. Otherwise we create ambiguity: Brgemm type suggests that there must be a repacking op, but the op may not in the graph (but may still be, who knows? - only the eliminate_brgemm_copy_b have this info, but it won't tell anyone)

As one option, we can create a bit-based enum that creates basic states, like stand_alone, with_repacking, with_compensations, with_scratchpad, external_repacking (another example [here](https://stackoverflow.com/a/4920859)). Then, we'll be able to mix this states into allowed combinations, like `with_repacking | with_scratchpad`.
When we move the repacking outside, we just going to set new brgemm type like this: `new_brgemm_type = old_brgemm_type & external_repacking`.
This way the Brgemm op would "know" exactly if there should be a certain operation in the graph (e.g. BrgemmCopyB), or it's not needed (or will be executed externally).

Reply:
Thanks for the idea, I reflected it in 157339 ticket